src/os/bluestore/BlueStore.cc

```diff
@@ -14725,6 +14728,31 @@ void BlueStore::_do_write_big_apply_deferred(
   }
 }
 
+void BlueStore::_detect_zeros_and_write_big(
+    uint64_t loffs,
+    BlobRef b,
+    uint64_t blob_len,
+    uint64_t o,
+    bufferlist& bl,
+    uint64_t o0,
+    uint64_t len0,
+    bool _marked_unused,
+    bool _new_blob,
+    WriteContext *wctx) {
+    if (!bl.is_zero()) {
+      // There is relevant data in this bufferlist, so write to disk
+      wctx->write(loffs, b, blob_len, o, bl, o0, len0, _marked_unused, _new_blob);
+      dout(20) << __func__ << " schedule write big: 0x"
+        << std::hex << loffs << "~" << blob_len << std::dec
+        << (_new_blob ? " new " : " reuse ")
+        << *b << dendl;
+      logger->inc(l_bluestore_write_big_blobs);
+    } else {
+      logger->inc(l_bluestore_zero_block_count_big);
```

Comment:
it might be helpful to log relevant notification on bypassing zero block at dout(20) . The same applies to do_write_small